lib/src/app.dart

```diff
@@ -26,6 +27,48 @@ import 'credentials.dart';
 import 'native/realm_core.dart';
 import 'user.dart';
 
+final _pinnedClient = () {
```

Comment:
This is not pinning the certificate, rather it's adding it to the set of trusted certificates. This doesn't prevent other certificates from being trusted by the app.

Reply:
It pins the root certificate. No other root certificates are allowed

Reply:
.. otherwise I should have done:
```dart
  final context = SecurityContext(withTrustedRoots: true);
```

Reply:
But only for the `HttpClient` we use ofcourse.

Reply:
`No other root certificates are allowed` this doesn't sound as the best approach imo.

Reply:
https://security.stackexchange.com/questions/224246/certficate-pinning-should-i-pin-the-leaf-or-intermediate#:~:text=By%20pinning%20against%20the%20root,not%20to%20mis%2Dissue%20certificates.

.. but it is trivial to change

Reply:
We restrict to only trust certificates signed (transitively) by ISRG X1 Root, which is less than what we trust today, but still a whole lot.. basically anything signed by lets-encrypt.

Reply:
.. and different from what is trusted on Android prior to 7.1.1

Reply:
After a long slack discussion I have added the root certificate instead of pinning it.

Reply:
This turns out to break it for @Navil. I was a bit quick to accept this change. Will revert to draft.